sentence_transformers/cross_encoder/losses/ListNetLoss.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+
+from sentence_transformers.cross_encoder import CrossEncoder
+
+
+class ListNetLoss(nn.Module):
+    """
+    ListNet loss for learning to rank. This loss function implements the ListNet ranking algorithm
+    which uses a list-wise approach to learn ranking models. It minimizes the cross entropy
+    between the predicted ranking distribution and the ground truth ranking distribution.
+    The implementation is optimized to handle padded documents efficiently by only processing
+    valid documents during model inference.
+
+    Args:
+        model (CrossEncoder): CrossEncoder model to be trained
+        eps (float): Small constant to prevent numerical instability in log.
+        pad_value (int): Value used for padding in variable-length document lists.
+            Documents with this value will be excluded from model inference for efficiency.
+
+    References:
+        - Learning to Rank: From Pairwise Approach to Listwise Approach: https://www.microsoft.com/en-us/research/publication/learning-to-rank-from-pairwise-approach-to-listwise-approach/
+        - `Training Examples > Learning to Rank <../../../examples/training/cross-encoder/training_ms_marco_ListNetLoss_v4.py>`_
+
+    Requirements:
+        1. Query with multiple documents (listwise approach)
+        2. Documents must have relevance scores/labels. Both binary and continuous labels are supported.
+        3. Variable-length document lists are handled through padding (pad_value)
+        4. Padded documents are automatically excluded from model inference for efficiency
+
+    Inputs:
+        +----------------------------------------+--------------------------------+
+        | Texts                                   | Labels                         |
+        +========================================+================================+
+        | (query, [doc1, doc2, ..., docN])       | [score1, score2, ..., scoreN]  |
+        +----------------------------------------+--------------------------------+
+        Note: Documents with label=pad_value are efficiently skipped during processing
+
+    Relations:
+        - /
+    """
+
+    def __init__(
+        self,
+        model: CrossEncoder,
+        eps: float = 1e-10,
+        pad_value: int = -1,
+        activation_fct: nn.Module | None = nn.Sigmoid(),
```

Comment:
We got better results with `nn.Identity` here right? Whatever worked best for this particular loss should be the default